src/build/buildApp.js

```diff
@@ -110,19 +109,14 @@ function maybeCopyScripts(srcs, dest) {
   });
 }
 
-function normalizeAppName(appName, url) {
-  // use a simple 3 byte random string to prevent collision
```

Comment:
What was this collision the original author was talking about? Not going to merge this before understanding the initial reason for it.

Reply:
https://www.azquotes.com/quote/935154

Good call.  I'm curious myself.

Reply:
Unfortunately, the [commit responsible for adding it](https://github.com/jiahaog/nativefier/commit/8f78dd03af47155c950dd6af8cef9599bd77f3cc) doesn't provide much context (to me).

Reply:
Can we resolve this as it is holding back a lot of app builds? @ronjouch open a separate issue to address your concern(s), please.

Reply:
@0xnu I think it’s a valid concern, unfortunately. To say nothing of the fact that @ronjouch is the project maintainer. If this PR is removing that functionality which came with a cautionary/explanatory comment, it would be prudent to understand the implications.

That being said, I can’t think of a reason for this. Unless there is some other identifier that’s used under the covers for preference files etc, to prevent collision. I do think it’s not that unreasonable to expect a user to have one copy of the app. In any case, if this needs to be a unique ID, we could certainly separate the presentation value (menu bar) from the internal identifier.

Reply:
:wave: @jiahaog . Do you remember why you put in place this _"simple 3 byte random string to prevent collision"_? A collision of what, when?

Reply:
> Can we resolve this as it is holding back a lot of app builds? @ronjouch open a separate issue to address your concern(s), please.

@0xnu, keep in mind that you can generate apps locally with this fix.  `node cli.js <all your configs here> "http://link.to.your.site" "/path/to/app/destination"` .  That's my workaround for now.

Reply:
@ronjouch The original commit that introduced the line is actually this, https://github.com/nativefier/nativefier/commit/7bfedc823b015c36f8500780a0fdbba1e6c45d55 to fix #21. 

The referenced issue mentions collisions of files in (Mac OS) AppData, so my understanding is that it's trying to prevent two separate apps with identical given names from sharing resources. The way Mac apps ordinarily work is that the developer specifies this ID, but in the absence of an explicit value Nativefier is attempting to create one.

I think there are two options to move forward:

- Accept that app names can collide and accept this MR in its current state (Do users really need to distinguish between 'Whatsapp' and 'WhatsApp'?)
- Add support for something like an `--id` argument that's used here, or fall back to the current behaviour if not specified.

Reply:
> The original commit that introduced the line is actually this: 7bfedc8 to fix #21.

That's right, thanks for finding it in the history. If I remember correctly, this was added because users might create two (or more) apps with the same URL / name. This will cause them to share the application data directory, which might not be intended by the user. For instance, you might create multiple apps for the same messaging platform, and use different accounts when logging into each app.

As such, removing the hash here is probably not the correct approach. Sorry, it's been quite some time since I've worked on this, but it would seem like a solution purely targeting the window name, but not how data is namespaced on disk would be desirable. Would that be possible? In that way we can also avoid introducing more flags to the already large API of Nativefier 🙂